main/define/src/mill/define/NamedParameterOnlyDummy.scala

```diff
@@ -0,0 +1,6 @@
+package mill.define
+
+/**
+ * Dummy class used to mark parameters that come after it as named only parameters
+ */
+object NamedParameterOnlyDummy
```

Comment:
perhaps a SIP idea: an annotation that forces callsites to use named parameters?

Reply:
or some new syntax at the declaration anyway

Reply:
Yeah that's a good idea!